statsmodels/regression/_prediction.py

```diff
@@ -157,7 +157,8 @@ def get_prediction(self, exog=None, transform=True, weights=None,
                 row_labels = None
 
         exog = np.asarray(exog)
-        if exog.ndim == 1 and (self.model.exog.ndim == 1 or
+        if exog.ndim == 1 and (self.model.exog is None or
```

Comment:
this doesn't work if exog is None, 
we need to check if exog had only 1 column

Reply:
The entire remove_data and still predict is fundamentally broken since this data isn't available when data has been removed.  I think probably must raise in this case.

Reply:
we don't need model.exog for predict
remove_data was designed for predict without carrying around all the original data

Reply:
Except that it is needed to handle edge cases like this.  Once it has been removed the correct decision cannot be reasoned here.

Reply:
As I mentioned on the mailing list, all we need to know is whether there is only 1 exog variable or more than 1.
It would be df_model except that doesn't count the const. In models without extra params, len(self.params) is sufficient

Reply:
Now using params since it needs to go through `np.dot`